src/Sylius/Bundle/CoreBundle/Telemetry/Collector/TechnicalDataCollector.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\CoreBundle\Telemetry\Collector;
+
+use Sylius\Component\Core\Telemetry\Collector\TelemetryDataCollectorInterface;
+use Sylius\Component\Core\Telemetry\DataProvider\DataProviderInterface;
+use Symfony\Component\Serializer\Normalizer\NormalizerInterface;
+
+/** @internal */
+final class TechnicalDataCollector implements TelemetryDataCollectorInterface
+{
+    /**
+     * @param iterable<DataProviderInterface> $dataProviders
+     */
+    public function __construct(
+        private iterable $dataProviders,
```

Comment:
Would be nice to have validation for the type inside, similar case in the other `iterable` arguments